core/primitives/src/crypto/signer.rs

```diff
@@ -53,15 +62,209 @@ impl KeyFile {
             panic!("Failed to write a key file {}", err);
         }
     }
+    // is_encrypted checks if current keyhash struct matches default one
```

Comment:
Doc comments should use three / in rust.

Consider using to rustfmt for code formatting.